bears/general/CasingBear.py

```diff
@@ -0,0 +1,153 @@
+            last_line = string_range.end.line
+            last_column = string_range.end.column
+        results.append(SourceRange.from_values(
+            text,
+            start_line=last_line,
+            start_column=last_column,
+            end_line=len(file),
+            end_column=len(file[-1])))
+
+        changes = {}
+        for result in results:
+            line_number = result.start.line
+            while line_number <= result.end.line:
+                start = 0 if line_number != result.start.line \
+                    else result.start.column
+                end = result.end.column if line_number == result.end.line \
+                    else len(file[line_number - 1])
+
+                line_changes = self.process(file[line_number - 1][start:end])
+                for prev in line_changes:
+                    changes[prev] = line_changes[prev]
+
+                line_number += 1
+
+        lines_changed = set()
+        result_texts = []
+        for prev in changes:
+            change = changes[prev]
+            diff = Diff(file)
+            first_line = -1
+            num_changes = 0
+            skip_var = False
+
+            for line_number, line in enumerate(file, start=1):
+                rep = re.sub("(?P<g1>[\\" + self.delim + "]*)" + prev +
+                             "(?P<g2>[\\" + self.delim + "]*)",
+                             "\g<g1>" + change + "\g<g2>", line)
+
+                if rep != line:
+                    if line_number in lines_changed:
+                        skip_var = True
+                    lines_changed.add(line_number)
+                    diff.change_line(line_number, line, rep)
+                    num_changes += 1
+                    if first_line == -1:
+                        first_line = line_number
+
+            if skip_var:
+                continue
+
+            msg = "Change '" + prev + "' to '" + change + "'"
+            if num_changes > 1:
+                msg += ": " + str(num_changes) + " lines affected"
+            result_texts.append(msg)
+            vio = "The following name change is suggested:"
+            vio += "".join("\n- " + string
+                           for string in result_texts)
+
+            yield Result.from_values(
+                self,
+                vio,
+                diffs={filename: diff},
+                file=filename,
+                line=first_line)
+            result_texts = []
+
+    def process(self, content):
+        """
+        Processes the string by splitting it based on delimiters
+        and then identifying the casing changes.
+
+        :param content: The string to be processed
+        :return:        A dict with current casing as key and
+                        proposed change as value
+        """
+        splits = re.split("[" + self.delim + "]", content)
+        results = {}
+        for split in splits:
+            if split not in self.keywords:
```

Comment:
not too sure about this approach, there are going to be imports, macros, converting everything doesnt seem sane